mcl-barrels.lua

```diff
@@ -0,0 +1,64 @@
+-- this bit of code modifies the mcl barrels to be compatible with
+-- pipeworks.
+
+-- Pipeworks Specific
+local tube_entry = "^pipeworks_tube_connection_wooden.png"
+
+-- Original Definitions
+local old_barrel = table.copy(minetest.registered_items["mcl_barrels:barrel_closed"])
+
+local groups = old_barrel.groups
+groups["tubedevice"] = 1
+groups["tubedevice_receiver"] = 1
+local groups_open = table.copy(groups)
+groups_open["not_in_creative_inventory"] = 1
+
+
+-- Override Construction
+local override_barrel = {
+   tiles = {"mcl_barrels_barrel_top.png^[transformR270",
+	    "mcl_barrels_barrel_bottom.png"..tube_entry,
+	    "mcl_barrels_barrel_side.png"..tube_entry},
+   after_place_node = function(pos, placer, itemstack, pointed_thing)
+      old_barrel.after_place_node(pos, placer, itemstack, pointed_thing)
+      pipeworks.after_place(pos, placer, itemstack, pointed_thing)
+   end,
+   tube = {
+      insert_object = function(pos, node, stack, direction)
+	 local meta = minetest.get_meta(pos)
+	 local inv = meta:get_inventory()
+	 return inv:add_item("main", stack)
+      end,
+      can_insert = function(pos, node, stack, direction)
+	 local meta = minetest.get_meta(pos)
+	 local inv = meta:get_inventory()
+	 if meta:get_int("splitstacks") == 1 then
+	    stack = stack:peek_item(1)
+	 end
+	 return inv:room_for_item("main", stack)
+      end,
+      input_inventory = "main",
+      connect_sides = {left = 1, right = 1, back = 1, front = 1, bottom = 1}
+   },
+   after_dig_node = function(pos, oldnode, oldmetadata, digger)
+      old_barrel.after_dig_node(pos, oldnode, oldmetadata, digger)
+      pipeworks.after_dig(pos)
+   end,
+   groups = table.copy(old_barrel.groups),
+   on_rotate = pipeworks.on_rotate
+}
```

Comment:
> formatting of this is pretty bad

how can I improve the format? maybe you think it's bad format because you uses a text editor different from mine

Reply:
I don't think the editor has anything to do with what he is pointing out.
My guess is that there are a few layers of tables within tables here.

You could improve readability by adding empty lines between the parts or by doing something like this:
```lua
local barrel_override = {}
barrel_override.tiles = {
    <tiles table content>
} -- << note the opening and closing brackets of multiline table on their own lines without content

barrel_override.after_place_node = function(<params>)
    <function statements>
end

....

barrel_override.on_rotate = pipeworks.on_rotate
```
(These are suggestions I personally prefer, not speaking for wsor4035. Many mods have the habbit of craming multilevel tables. This is not so easy to read and can easily lead to errors when maintaining the code)

Reply:
Format it like you did for the furnace override, that one looks ok.

Also, please use tabs for indentation, not spaces.